### PR TITLE
feat: merge and store balance data using FileManager

### LINKED ISSUE #153: Feat: Merge and store balance data using FileManager
## Context

After fetching new balances, they must be merged with existing data and persisted to storage. FileManager ensures atomic writes to prevent data corruption.

## Current Behavior

1. New balances are fetched and converted but not stored
2. Existing balance data is not updated with new values
3. Balance data is lost when process completes

## Expected Behavior

- Merge newly fetched balances with existing balance data (if any)
- Create proper BalanceData structure with metadata
- Include chain_id (42170 for Arbitrum Nova) in metadata
- Include reward_distributor address in metadata
- Use FileManager.saveBalanceData() to write atomically
- Ensure all dates are included in final output

## Required Context

- `src/services/file-manager.ts` - saveBalanceData() method usage
- `src/types/index.ts` - complete BalanceData type structure
- `src/services/distributor-detector.ts` - example of saving data with FileManager
- `docs/specs/balance-fetcher.md` - Section 7 output data structure

## Out of Scope

- Progress notifications or logging
- Partial data writes on failure
- Balance analytics or validation
- Cleanup of temporary data

### PR DESCRIPTION
**Pull request by yahgwai**
## Summary
- Implements balance data persistence after fetching using FileManager
- Merges newly fetched balances with existing data incrementally  
- Includes proper metadata (chain_id and reward_distributor address)

## Implementation Details
This PR implements issue #153 by modifying the `fetchBalances` method to save fetched balance data:

1. **Data Structure**: Creates proper `BalanceData` structure with metadata containing:
   - `chain_id`: 42170 (ARBITRUM_NOVA)
   - `reward_distributor`: The distributor's address

2. **Incremental Merge**: Loads existing balance data and merges with new balances before saving

3. **Atomic Writes**: Uses `FileManager.writeDistributorBalances()` for atomic file updates

4. **Multiple Distributors**: Processes and saves data for all fetched distributors

## Testing
Comprehensive unit tests added in `balance-fetcher-persistence.test.ts` covering:
- Basic storage functionality
- Metadata creation
- Incremental merge behavior  
- Multiple distributor handling
- Edge cases (no new data to fetch)

All tests pass ✅

## TDD Process
Followed strict TDD methodology:
1. **RED**: Wrote failing tests for all scenarios
2. **GREEN**: Implemented minimal code to pass tests
3. **REFACTOR**: Extracted `createBalanceData` helper method for clarity

Closes #153

### CODE REVIEW

File: src/balance-fetcher.ts [hunk 1]
```diff
@@ -19,6 +19,42 @@ export class BalanceFetcher {
...
+   */
+  private createBalanceData(
+    address: string,
+    existingData: BalanceData | undefined,
+    newBalances: Record<string, string>,
+    allFetches: Array<{ address: string; date: string; block: number }>,
+  ): BalanceData {
+    const balanceData: BalanceData = {
+      metadata: {
+        chain_id: CHAIN_IDS.ARBITRUM_NOVA,
```
Comment (yahgwai) [inline src/balance-fetcher.ts]: if there is existing balance data then the chain id should already be set. Also, we never want to use arbnova as the chain id - always we want to get it from the provider.
